File: algorithms/swift/704-binary-search.swift
// https://leetcode.com/problems/binary-search/submissions/

class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        func binarySearch(_ l: Int, _ r: Int) -> Int {
            if (l > r) { 
                return -1
            }
            let mid: Int = (l+r)/2
            if (nums[mid] == target) { 
                return mid
            }
            if (nums[mid] > target) {
                return binarySearch(l, mid-1)
            }
            return binarySearch(mid+1, r)
        }
        return binarySearch(0, nums.count-1)
    }
}

File: algorithms/c++/704-binary-search.cpp
// https://leetcode.com/problems/binary-search/
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int binarySearch(vector<int>& nums, int& target, int left, int right) {
        if (left > right) return -1;
        int mid = (left + right)/2;
        if (nums[mid] == target) return mid;
        if (nums[mid] > target) return binarySearch(nums, target, left, mid-1);
        return binarySearch(nums, target, mid+1, right);
    }
    
    int search(vector<int>& nums, int target) {
        return binarySearch(nums, target, 0, nums.size()-1);
    }
};

int main() {

    return 0;
}